openghg/standardise/surface/_openghg.py

```diff
@@ -69,170 +71,168 @@ def parse_openghg(
-    network_case_options = [network, network.upper(), network.lower()]
-
-    # Extract centralised data for site (if present)
-    site_data = get_site_info(site_filepath=site_filepath)
-    for site_value in site_case_options:
-        if site_value in site_data:
-            site_info_all = site_data[site_value]
-            break
-    else:
-        logger.info(
-            "Unknown site. Will attempt to extract metadata from dataset attributes or input keywords"
-        )
-        site_info_all = {}
-
-    for network_value in network_case_options:
-        if network_value in site_info_all:
-            site_info = site_info_all[network_value]
-            break
-    else:
-        logger.info(
-            "Network {network} does not match with site {site}. Will attempt to extract metadata from dataset attributes or input keywords"
-        )
-        site_info = {}
-
-    if site_info:
-        # Ensure keywords match to metadata names for station values
-        # e.g. "station_longitude" derived from "longitude"
-        for key in attributes_needed:
-            prefix = "station_"
-            if key.startswith(prefix):
-                split_key = key.split("_")[1:]
-                short_key_option1 = "_".join(split_key)
-
-                split_key.insert(1, "station")  # to catch "height_station_masl"
-                short_key_option2 = "_".join(split_key)
-
-                short_key_options = [short_key_option1, short_key_option2]
-                for short_key in short_key_options:
-                    if short_key in site_info:
-                        site_info[key] = site_info[short_key]
-                        break
-
-    # Load attributes data for network if present
-    param_data = load_internal_json(filename="attributes.json")
-    for network_value in network_case_options:
-        if network_value in param_data:
-            network_params = param_data[network_value]
-            site_attributes = network_params["global_attributes"]
-            break
-    else:
-        site_attributes = {}
-
-    # Define sources of attributes to use when defining metadata
-    # The order here influences the hierarchy if keys appear multiple times.
-    # kwargs allow additional variables such as "station_longitude" to be included if needed.
-    # 2023-04: Re-ordered to be - values explicitly passed, stored data (2), attributes from dataset
-    # # attribute_sources = [attributes, kwargs, site_info, site_attributes]
-    attribute_sources = [kwargs, site_info, site_attributes, attributes]
-
-    # Search attributes sources (in order) and populate metadata
-    for param in attributes_needed:
-        for source in attribute_sources:
-            if param in source:
-                metadata[param] = source[param]
+    with get_dataset(dataset=dataset, filepath=filepath) as data:
+
+        # Extract current attributes from input data
+        attributes = data.attrs
```

Comment:
Something like [build_surface_metadata](https://github.com/openghg/openghg/blob/461a15a14ec1173f596f5f1b126cc4fb64e1b82f/openghg/util/_metadata_util.py#L399) from the other prototype could still be incorporated into this workflow. May be something we could approach on a separate PR though as this could also make use of other pre-existing functions like [`merge_dict`](https://github.com/openghg/openghg/blob/devel/openghg/util/_metadata_util.py#L175).